openeo_processes_dask/process_implementations/cubes/aggregate.py

```diff
@@ -109,11 +124,140 @@ def aggregate_temporal_period(
     )
 
 
+def _aggregate_geometry(
+    data: RasterCube,
+    geom,
+    transform,
+    reducer: Callable,
+):
+    data_dims = list(data.dims)
+    y_dim = data.openeo.y_dim
+    x_dim = data.openeo.x_dim
+    t_dim = data.openeo.temporal_dims
+    t_dim = None if len(t_dim) == 0 else t_dim[0]
+    b_dim = data.openeo.band_dims
+    b_dim = None if len(b_dim) == 0 else b_dim[0]
+
+    y_dim_size = data.sizes[y_dim]
+    x_dim_size = data.sizes[x_dim]
+
+    # Create a GeoSeries from the geometry
+    geo_series = gpd.GeoSeries(geom)
+
+    # Convert the GeoSeries to a GeometryArray
+    geometry_array = geo_series.geometry.array
+
+    mask = rasterio.features.geometry_mask(
+        geometry_array, out_shape=(y_dim_size, x_dim_size), transform=transform
+    )
+
+    if t_dim is not None:
+        mask = np.expand_dims(mask, axis=data_dims.index(t_dim))
+    if b_dim is not None:
+        mask = np.expand_dims(mask, axis=data_dims.index(b_dim))
+
+    masked_data = data * mask
```

Comment:
Hey, I got here from https://github.com/xarray-contrib/xvec/pull/52 where @masawdah opened a PR with a variation of this. 

The same code as is here was there as well but I believe it is wrong. `mask` is a boolean array with False where the `geometry_array` has been burned in, the rest is True. Line 159 results in an array where all the values are preserved except those where the geometry is, where the values are now 0 (not missing!).

Any reduction of this masked array it therefore likely wrong. 

Even if you passed `invert=True` to `rasterio.features.geometry_mask` to get True indicating the location of geometry and False the area that should be taken away, the line 159 still keeps 0 where the geometry is not, therefore mean, median etc will be wrong. You would need to create a mask resulting in missing values here I believe.

I may be wrong as this is clearly reviewed PR but I don't think that is the case. The tests here also do not verify any correctness of values and this code indeed produces correctly formed cube. Just with completely wrong results.

I have refactored the logic in that `xvec` PR and verified the result against other implementation so you can check how it is done there. Or we may be able to get the built-in functionality of xvec to the state where you can use it directly, if you dare to depend on version 0.2.0 of it :).

Reply:
@martinfleis  Thanks for pointing this out. Actually, this work was done simultaneously with [xarray-contrib/xvec#52](https://github.com/xarray-contrib/xvec/pull/52) . The intention behind it was to replace this effort with a built-in functionality of xvec once it is ready.  

I saw your refactoring for the implementation, and it looks great. Let's double-check once again to confirm the best :)